engines/opensearch/config.py

```diff
@@ -0,0 +1,222 @@
+from datetime import date
+import os
+import copy
+
+OPENSEARCH_HOST = os.getenv("AIPS_OPENSEARCH_HOST") or "aips-opensearch"
+OPENSEARCH_PORT = os.getenv("AIPS_OPENSEARCH_PORT") or "9200"
+OPENSEARCH_URL = f"http://{OPENSEARCH_HOST}:{OPENSEARCH_PORT}"
+
+def base_field(type, **kwargs):
+    return {"type": type, "store": True} | kwargs
+
+def text_field(**kwargs):
+    return base_field("text", **kwargs)
+
+def boolean_field():
+    return base_field("boolean")
+
+def double_field():
+    return base_field("double")
+
+def integer_field():
+    return base_field("integer")
+
+def keyword_field(**kwargs):
+    return base_field("keyword", **kwargs)
+
+# {"multiValued": "true", "docValues": "true"}
+def date_field():
+    return base_field("date")
+
+def basic_schema(field_mappings, id_field="_id"):
+    return {
+        "id_field": id_field,
+        "schema": {"mappings": {"properties": field_mappings}},
+    }
+
+def body_title_schema():
+    return basic_schema({"title": text_field(), "body": text_field()})
+
+def dense_vector_schema(
+    field_name, dimensions, similarity_score, quantization_size, additional_fields={}
```

Comment:
Is there support for quantization_size in opensearch? Future addition?

Reply:
I've gone back and read Opensearch's documentation for quantization and it still requires a field for each quantization size/type. I'll give you a more complete synopsis of Opensearch's quantization features on our next sync